### DISCUSSION TITLE
Editing permissions for multiple collections at once (bulk update)

### DISCUSSION THREAD

**Discussion by log4shell** [directus/directus#11052] [Feature Requests]
The current UI for permission settings only allows me to edit one permission for one collection at a time, which turns out to be a very inefficient and repetitive piece of manual work.

What I want to do is to configure permissions so that CRUD operations are allowed for different user roles depending on the value of the "status" field in the entity, e.g., "Author" can create and read any item but only update those in "draft" status, whereas "Editor" can also update and publish those in "review", but once they're published only the "Content manager" role can make further changes, etc.

All of this is possible in Directus, but it's a pain to implement due to the fact I have to repeat the same steps for each permission type and on each collection type, which doesn't scale well. In my small proof-of-concept scenario, I have about 10 different entities plus double the amount of M2M joining collections for a total of 30 collections, which means I have to do 30 (collections) * 4 (CRUD permissions) * 3 (roles) = 360 repetitive steps just for a small subset of our anticipated data model. I know it's possible to copy/paste the raw value of the custom permission rule to reduce the manual steps for each entry a little, but the overall number of edits needed is still the same.

Give that most, if not all, of my collections containing the status field should follow the same permission rule related to it, I would love to see some kind of 'bulk edit' mode where I can select all collections for which I want to set a specific permission and then update them all at once - similar to how it's possible to edit the content of multiple entities within a collection. This would drastically reduce the number of permission edits I have to do. Thanks!

**Comment by benhaynes** [comment]
I agree, and we can certainly add some helpers to make common actions easier/faster. Also, it would be nice to support bulk permission changes... keeping in mind that the difference in data model between collections could lead to some unexpected errors on the admin side.

I wonder if exposing some copy/paste actions at the permission level would help? Could be a right click context menu on the toggle... or something like that.

**Reply by log4shell** [reply]
Yes, I know that bulk updates of custom permissions based on collection fields can be tricky but it should be possible to limit such edits to the fields that all the selected collections have in common.

Another thing that would be helpful is to have some kind of "cloning" or copy/paste of permission from one collection to another, e.g., if I'm adding a new collection for which I want to apply the same permission model, I would like to clone it from an existing collection's permission - or perhaps even have a some type of abstract permission groups so that all collections I put under a certain group inherits the same set of permissions.

I know this could add a bit of complexity, so finding the right balance between efficient permission management and simplicity is the key.

**Reply by benhaynes** [reply]
Exactly.  One option is to remove barriers from Directus  and put the onus on developers to ensure a set permissions based on their own data model. Essentially we would just error out on misconfiguration...  but support easier cloning of permissions, regardless of the fields within the collection.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
